src/cpp/src/image_generation/models/unet_inference.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/image_generation/unet2d_condition_model.hpp"
+
+namespace ov {
+namespace genai {
+
+class UNet2DConditionModel::UNetInference {
+
+public:
+    virtual void compile(std::shared_ptr<ov::Model> model, const std::string& device, const ov::AnyMap& properties) = 0;
+    virtual void set_hidden_states(const std::string& tensor_name, ov::Tensor encoder_hidden_states) = 0;
+    virtual void set_adapters(AdapterController& adapter_controller, const AdapterConfig& adapters) = 0;
+    virtual ov::Tensor infer(ov::Tensor sample, ov::Tensor timestep) = 0;
+
+    // utility function to resize model given optional dimensions.
+    static void reshape(std::shared_ptr<ov::Model> model,
+                        std::optional<int> batch_size = {},
+                        std::optional<int> height = {},
+                        std::optional<int> width = {},
+                        std::optional<int> tokenizer_model_max_length = {})
+    {
+        std::map<std::string, ov::PartialShape> name_to_shape;
+        for (auto&& input : model->inputs()) {
+            std::string input_name = input.get_any_name();
+            name_to_shape[input_name] = input.get_partial_shape();
+            if (input_name == "timestep") {
+                name_to_shape[input_name][0] = 1;
+            } else if (input_name == "sample") {
+                if (batch_size) {
+                    name_to_shape[input_name][0] = *batch_size;
```

Comment:
in case of SD 1.x, 2.x (non LCM), within a sample we will reshape to batch 2 (because of classifier free guidance), so calling `reshape(model, 1)` from NPU impl will not set batch size to 1